Editor/PropertyDatabase/PropertyDatabaseWindow.cs

```diff
@@ -0,0 +1,342 @@
+        var window = GetWindow<PropertyDatabaseWindow>();
+        window.titleContent.text = "Property Database";
+
+        return window;
+    }
+
+    public void OnEnable()
+    {
+        if (s_PropertyDatabaseWindowUI == null)
+            CreatePropertyDatabaseWindowUI();
+
+        rootVisualElement.Add(s_PropertyDatabaseWindowUI);
+    }
+
+    public void OnDestroy()
+    {
+        if (s_PropertyDatabaseWindowUI != null)
+        {
+            s_PropertyDatabaseWindowUI.Clear();
+            s_PropertyDatabaseWindowUI = null;
+        }
+
+        s_PropertyDatabaseFileName = string.Empty;
+        s_PropertyDatabaseView = default(PropertyDatabaseView);
+        s_CreatedFromPropertyDatabaseView = false;
+    }
+
+    private static VisualTreeAsset LoadAssetPath(string filePath)
+    {
+        var UIAsset = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(filePath);
+        if (UIAsset == null)
+            Debug.Log($"Could not load the { filePath } file.");
+
+        return UIAsset;
+    }
+
+    private static void CreatePropertyDatabaseWindowUI()
+    {
+        var windowUIAsset = LoadAssetPath(k_PropertyDatabaseWindowAssetPath);
+        if (windowUIAsset == null)
+            return;
+
+        s_PropertyDatabaseWindowUI = windowUIAsset.Instantiate(null);
+        var header = s_PropertyDatabaseWindowUI.Q("Header");
+        var headerText = header.Q<Label>();
+        headerText.text = "Load any Property Database file to visualize its content.";
+
+        var loadButton = s_PropertyDatabaseWindowUI.Q<Button>("LoadButton");
+        var refreshButton = s_PropertyDatabaseWindowUI.Q<Button>("RefreshButton");
+        loadButton.clicked += LoadPropertyDatabaseFilePath;
+        refreshButton.clicked += ShowPropertyDatabaseListView;
+    }
+
+    private static void LoadPropertyDatabaseFilePath()
+    {
+        var initialFolder = Path.GetDirectoryName(Application.dataPath);
+        var filePath = EditorUtility.OpenFilePanel("Open Property Database", initialFolder, "db");
+
+        if (string.IsNullOrEmpty(filePath))
+            return;
+        else if (!File.Exists(filePath + ".st"))
+        {
+            Debug.Log($"The file you are trying to load { filePath } is not a PropertyDatabase.");
+            return;
+        }
+
+        var propertyDatabase = new PropertyDatabase(filePath, true);
+        if (propertyDatabase == null)
+        {
+            Debug.Log($"Could not load the { filePath } file.");
+            return;
+        }
+
+        s_CreatedFromPropertyDatabaseView = false;
+        s_PropertyDatabaseView = propertyDatabase.GetView();
+        s_PropertyDatabaseFileName = Path.GetFileName(filePath);
+        ShowPropertyDatabaseListView();
+    }
+
+    private static void ShowPropertyDatabaseListView()
+    {
+        if (s_PropertyDatabaseView.Equals(default(PropertyDatabaseView)))
+            return;
+
+        if (s_PropertyDatabaseWindowUI == null)
+            s_PropertyDatabaseWindowUI.Clear();
```

Comment:
This will throw a NullReferenceException. Should this be a return instead?

Reply:
Yes, absolutely. Thanks for the catch, I will change that!